packages/labs/eleventy-plugin-lit/src/util/render-in-worker-thread.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {parentPort} from 'worker_threads';
+import {render} from '@lit-labs/ssr/lib/render-with-global-dom-shim.js';
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+
+import type {Message} from './types';
+
+if (parentPort === null) {
+  throw new Error(
+    'render-in-worker-thread.js must only be run in a worker thread'
+  );
+}
+
+parentPort.on('message', async (message: Message) => {
+  switch (message.type) {
+    case 'initialize-request': {
```

Comment:
Add a guard that only one `initialize-request` can ever be sent for the lifetime of a worker?